src/components/EmailVerification/types.ts

```diff
@@ -0,0 +1,14 @@
+const resultTypeOfEmail = {
+  pause: 'pause', // 5분간 재전송 금지,
+  overSending: 'overSending', //이메일 인증 횟수 초과
+  duplicate: 'duplicate', //이메일 중복
+  success: 'success',
+  serverError: 'serverError',
+} as const;
+
+export type ResultTypeOfEmail = keyof typeof resultTypeOfEmail;
```

Comment:
해당 객체는 실제로 쓰이지 않으므로 다음과 같이 사용하는 것이 좋아보입니다.
```suggestion
// pause 5분간 재전송 금지
// overSending 이메일 인증 횟수 초과
// duplicate 이메일 중복
export type ResultTypeOfEmail = 'pause' | 'overSending' | 'duplicate' | 'succes' | 'serverError';
```

Reply:
저번에 enum방식을 사용해보는 걸 추천받아서 사용해봤습니다

Reply:
enum은 상수 값에 의미를 부여하거나 참조를 통해서 값을 접근하여 의미를 묶어주는 역할을 수행합니다.

여기서는 키와 값이 같으니 값에 의미를 부여하는 방식은 의미가 없을 것 같고 그렇다면 해당 타입들을 `resultTypeOfEmail`에 대한 값들로 묶어주는 역할을 하는데(resultTypeOfEmail 에는 pause, overSending, duplicate 등이 있다~.) 실제로 사용하는 측에서는 값들에 직접 접근을 하고 있어서 enum 사용의 의미가 없어보입니다.

![image](https://user-images.githubusercontent.com/19388573/231333561-1ccce4a9-934e-4dc1-bdcd-b4dfa31d133a.png)

enum 방식으로 사용하시려면 객체를 export하여 사용하는 측에서 활용하는 것이 좋아보입니다.
```
// src/components/EmailVerification/types.ts
export const resultTypeOfEmail = {
  pause: 'pause', // 5분간 재전송 금지,
  overSending: 'overSending', //이메일 인증 횟수 초과
  duplicate: 'duplicate', //이메일 중복
  success: 'success',
  serverError: 'serverError',
} as const;

export type ResultTypeOfEmail = keyof typeof resultTypeOfEmail;


// src/components/EmailVerification/index.tsx
import { ResultOfEmailAPI, resultTypeOfEmail } from './types';

const onClickEmailBtn = async () => {
    try {
      const result = await sendVerificationEmail();
      switch (result.type) {
        case resultTypeOfEmail.duplicate:
          setEmail((prev: InputDataType) => ({
            ...prev,
            errorMsg: '이미 회원가입된 이메일이에요.',
          }));
          break;
        case resultTypeOfEmail.pause:
          setOpenToastModal(false);
          setAlertModalChilde(childOfModalForPause);
...

```

Reply:
enum 을 사용하는 방향을 코드 수정했습니다.